### PR TITLE
Preallocate output string of str.title() with required size

### PR DESCRIPTION
**Pull request by Jongy**

### CODE REVIEW

File: vm/src/obj/objstr.rs [hunk 1]
```diff
@@ -425,7 +425,7 @@ impl PyString {
     /// uppercase character and the remaining characters are lowercase.
     #[pymethod]
     fn title(&self, _vm: &VirtualMachine) -> String {
-        let mut title = String::new();
+        let mut title = String::with_capacity(self.value.len());
```
Comment (adrian17) [inline vm/src/obj/objstr.rs]: Could we also do the same in `__mul__`?

Comment (Jongy) [reply vm/src/obj/objstr.rs]: Yeah, and in a few other spots as well. I have it ready but I intended to do the other batch in another PR.

Comment (Jongy) [reply vm/src/obj/objstr.rs]: Ehh I'll just add another commit

Comment (Jongy) [reply vm/src/obj/objstr.rs]: Actually, it gets a bit messy so I'll do that in another PR

**Pull request merged**